cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1604,6 +1634,17 @@ impl RegMemImm {
             RegMemImm::Imm { .. } => {}
         }
     }
+
+    fn map_src_reg_as_def<RUM: RegUsageMapper>(&mut self, mapper: &RUM) {
```

Comment:
What do you think of `map_as_def`? I don't feel too strongly about that but I think it would pretty helpful to explain here what is going on (e.g. we have to pretend the register is writable because `map_def` only accepts writable registers).

Reply:
I like it!